examples/bri-3/src/bri/merkleTree/capabilities/updateMerkleTree/updateMerkleTreeCommand.handler.ts

```diff
@@ -0,0 +1,37 @@
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+import { MerkleTreeAgent } from '../../agents/merkleTree.agent';
+import { MerkleTreeStorageAgent } from '../../agents/merkleTreeStorage.agent';
+import { UpdateMerkleTreeCommand } from './updateMerkleTree.command';
+import { BpiMerkleTree } from '../../models/bpiMerkleTree';
+
+@CommandHandler(UpdateMerkleTreeCommand)
+export class UpdateMerkleTreeCommandHandler
+  implements ICommandHandler<UpdateMerkleTreeCommand>
+{
+  constructor(
+    @InjectMapper() private mapper: Mapper,
+    private readonly agent: MerkleTreeAgent,
+    private readonly storageAgent: MerkleTreeStorageAgent,
+  ) {}
+
+  async execute(command: UpdateMerkleTreeCommand) {
+    const merkleTreeUpdateCandidate: BpiMerkleTree =
+      await this.agent.fetchMerkleTreeCandidateByIdAndThrowIfValidationFails(
+        command.id,
+      );
+
+    this.agent.updateMerkleTree(
+      merkleTreeUpdateCandidate,
+      command.leaves,
+      command.hashFunction,
+    );
+
+    const newMerkleTree = await this.storageAgent.updateMerkleTree(
```

Comment:
updateMerkleTree => storeUpdatedMerkleTree?

Reply:
I will rename BUT! storageAgent.storeUpdatedMerkleTree is redundant. There is no confusion around storageAgent.updateFoo(). storageAgent.storeUpdateFoo() offers no extra information.  Even if you make the argument to sync all storageAgent calls so they are all storageAgent.store....()  What is the point of naming the agent storageAgent?  it will always be doing DB/storage actions.    I think we should make naming scheme decisions for a good reason rather than pure alignment. i.e. the shortest name that uniquely and correctly identifies the usage.